src/picongpu/include/algorithms/KinEnergy.hpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2017 Heiko Burau, Axel Huebl
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "simulation_defines.hpp"
+#include "algorithms/Gamma.hpp"
+
+namespace picongpu
+{
+
+using namespace PMacc;
+
+/** Computes the kinetic energy of a particle given its momentum and mass.
+ *
+ * The mass may be zero.
+ *
+ * For massive particle with low energy the non-relativistic
+ * kinetic energy expression is used in order to avoid bad roundings.
+ *
+ */
+template< typename precisionType = float_X >
+struct KinEnergy
+{
+    typedef precisionType valueType;
```

Comment:
please use capital letters for the begin of a type of `precisionType` and `valueType`